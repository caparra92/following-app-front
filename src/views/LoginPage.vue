<template>
  <ion-page>
    <ion-content>
        <ion-grid class="container">
          <ion-row>
            <ion-col>
              <div class="multi-button">
                    <button class="multi-button_button toggle" disabled>Login</button>
                    <button class="multi-button_button toggle" @click="changeView">Register</button>
                  </div>
            </ion-col>
          </ion-row>
          <ion-row>
              <ion-col><p class="title">Login</p></ion-col>
            </ion-row>
            <form action="">
              <ion-list>
              <ion-row>
                <ion-col>
                  <ion-item>
                    <ion-input class="has-focus" v-model="email">
                      <ion-icon size="large" :icon="personCircleOutline"></ion-icon>
                    </ion-input>
                  </ion-item>
                </ion-col>
              </ion-row>
              <ion-row>
                <ion-col>
                  <ion-item>
                    <ion-input type="password" v-model="password">
                      <ion-icon size="large" :icon="lockClosedOutline"></ion-icon>
                    </ion-input>
                  </ion-item>
                </ion-col>
              </ion-row>
              <ion-row>
                <ion-col>
                  <ion-item>
                    <ion-button @click="login">Enter</ion-button>
                  </ion-item>
                </ion-col>
              </ion-row>
              </ion-list>
            </form>
        </ion-grid>
    </ion-content>
  </ion-page>
</template>

<script setup lang="ts">
import router from '../router';
import { IonPage, IonContent, IonList, IonItem, IonInput, IonIcon, IonButton, IonGrid, IonRow, IonCol } from '@ionic/vue';
import { lockClosedOutline, personCircleOutline } from 'ionicons/icons';
import { useStore } from '../stores/store';

const store = useStore();
const email = null;
const password = null;

console.log(`Email: ${email}, password: ${password}`)

const changeView = () => {
  router.push("Register");
};

const login = async() => {
  await store.login(email, password);
}

</script>

<style scoped>

.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  height: 100vh;
  text-align: center;
}

.multi-button {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 0 1.2em;
  }
  
  .toggle {
    margin: 2em 0;
    width:5em;
    height: 2.3em;
    align-self: center;
    font-family: 'Outfit', sans-serif;
    text-transform: capitalize;
    width: 100%;
    height: 4em;
    margin: 1.2em 1.2em 0 1.2em;
    font-size: 1em;
    border-radius: 7px;
  }

  .toggle:first-child {
    border-radius: 7px 0 0 7px !important;
    margin-right: 0;
    background: rgba(var(--ion-color-primary-rgb), .6);
    color: var(--ion-color-light);
  }

  .toggle:last-child {
    border-radius: 0 7px 7px 0 !important;
    margin-left: 0;
    background: rgba(var(--ion-color-primary-rgb), 1);
    color: var(--ion-color-light);
  }

.title {
  font-family: 'Outfit', sans-serif;
  color: var(--ion-color-primary);
  font-size: 30px;
  font-weight: 400;
}

ion-input {
  background: var(--ion-color-secondary);
  color: var(--ion-color-primary);
  height: 4em;
  border-radius: 7px;
  outline: none;
  margin: 1.2em;
  transition: all 0.30s ease-in-out;
}

ion-input:focus {
  box-shadow: 0 0 5px rgba(var(--ion-color-primary-rgb), 1);
  padding: 3px 0px 3px 3px;
  margin: 5px 1px 3px 0px;
  border: 1px solid rgba(var(--ion-color-primary-rgb), 1);
}

ion-icon {
  position: absolute;
  right: 0;
  padding: .4em;
}

ion-button {
  font-family: 'Outfit', sans-serif;
  text-transform: capitalize;
  width: 100%;
  height: 4em;
  margin: 1.2em 1.2em 0 1.2em;
  font-size: 1em;
  border-radius: 7px;
}

</style>