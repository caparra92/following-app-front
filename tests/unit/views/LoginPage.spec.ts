import { mount, shallowMount } from '@vue/test-utils';
import LoginPage from '@/views/LoginPage.vue';

describe('Login tests', () => {
    test('Verify HTML structure', () => {
        const wrapper = mount(LoginPage);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Verify if a button contains a text', () => {
        const wrapper = mount(LoginPage);
        const button = wrapper.findComponent('ion-button');
        expect(button.text()).toBe('Enter');
    });
})