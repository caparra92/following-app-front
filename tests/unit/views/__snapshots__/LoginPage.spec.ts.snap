// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests Verify HTML structure 1`] = `
<div class="ion-page">
  <ion-content class="">
    <ion-grid class="container">
      <ion-row class="">
        <ion-col class="">
          <div class="multi-button"><button class="multi-button_button toggle" disabled="">Login</button><button class="multi-button_button toggle">Register</button></div>
        </ion-col>
      </ion-row>
      <ion-row class="">
        <ion-col class="">
          <p class="title">Login</p>
        </ion-col>
      </ion-row>
      <form action="">
        <ion-list class="">
          <ion-row class="">
            <ion-col class="">
              <ion-item class="">
                <ion-input class="has-focus">
                  <ion-icon></ion-icon>
                </ion-input>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row class="">
            <ion-col class="">
              <ion-item class="">
                <ion-input class="">
                  <ion-icon></ion-icon>
                </ion-input>
              </ion-item>
            </ion-col>
          </ion-row>
          <ion-row class="">
            <ion-col class="">
              <ion-item class="">
                <ion-button class="">Enter</ion-button>
              </ion-item>
            </ion-col>
          </ion-row>
        </ion-list>
      </form>
    </ion-grid>
  </ion-content>
</div>
`;
